src/main/java/com/desiato/puresynth/services/UserService.java

```diff
@@ -42,8 +50,17 @@ public void deleteUser(Long id) {
     }
 
     public User createUser(String email, String password) {
-        User newUser = new User(email, password);
-        userRepository.save(newUser);
-        return newUser;
+        User newUser = new User(email, projectConfig.passwordEncoder().encode(password));
+        return userRepository.save(newUser);
+    }
+
+    public User updateUserOrThrow(Long id, UserRequestDTO userRequestDTO) {
```

Comment:
I would just name it `updateUser`. The `orThrow` make sense when you retrieve elements, as it shows your intention what happens if the element won't be found.